httpie/utils.py

```diff
@@ -1,19 +1,72 @@
 import json
 import mimetypes
+import re
 import time
 from collections import OrderedDict
 from http.cookiejar import parse_ns_headers
 from pprint import pformat
-from typing import List, Optional, Tuple
-import re
+from reprlib import recursive_repr
+from typing import Any, List, Optional, Tuple
 
 import requests.auth
 
 RE_COOKIE_SPLIT = re.compile(r', (?=[^ ;]+=)')
+Item = Tuple[str, Any]
+Items = List[Item]
+
 
+class JsonDictPreservingDuplicateKeys(OrderedDict):
+    """A specialized JSON dict preserving duplicate keys.
+
+    Note: Python 3.6 & 3.7 do not support duplicate keys sorting.
+          See https://bugs.python.org/issue23493#msg400929.
+
+    """
```

Comment:
What do you mean exactly by "duplicate key sorting"?

Reply:
I see that you check the Python version in the tests and refer from there to this docstring. It would be better to add a constant and explain everything in a comment next to it instead of the version checks and repeated explanations.

```python
JsonDictPreservingDuplicateKeys:
    # …
    SUPPORTS_SORTING = sys.version_info >= (3, 8)
```

Reply:
A minor thing: let’s wrap all URLs in comments in angled braces like this:

```
See <https://bugs.python.org/issue23493#msg400929>.
```

That makes URLs visually stand out, easier to copy, and harder to accidentally change, especially when used in a sentence like above.